// ts-devrant/dist/types.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoteState = exports.CollabState = exports.Action = exports.NotificationKind = exports.RantType = exports.PostType = void 0;
__exportStar(require("./response-types"), exports);
var PostType;
(function (PostType) {
    PostType["Rant"] = "rant";
    PostType["JokeMeme"] = "joke/meme";
    PostType["Question"] = "question";
    PostType["DevRant"] = "devRant";
    PostType["Random"] = "random";
    PostType["Collab"] = "collab";
    PostType["Undefined"] = "undefined";
})(PostType = exports.PostType || (exports.PostType = {}));
var RantType;
(function (RantType) {
    RantType[RantType["Rant"] = 1] = "Rant";
    RantType[RantType["Collab"] = 2] = "Collab";
    RantType[RantType["JokeMeme"] = 3] = "JokeMeme";
    RantType[RantType["Question"] = 4] = "Question";
    RantType[RantType["DevRant"] = 5] = "DevRant";
    RantType[RantType["Random"] = 6] = "Random";
    RantType[RantType["Undefined"] = 7] = "Undefined";
})(RantType = exports.RantType || (exports.RantType = {}));
var NotificationKind;
(function (NotificationKind) {
    NotificationKind["CommentContent"] = "comment_content";
    NotificationKind["CommentDiscuss"] = "comment_discuss";
    NotificationKind["CommentMention"] = "comment_mention";
    NotificationKind["CommentVote"] = "comment_vote";
    NotificationKind["ContentVote"] = "content_vote";
    NotificationKind["CantSub"] = "rant_sub";
})(NotificationKind = exports.NotificationKind || (exports.NotificationKind = {}));
var Action;
(function (Action) {
    Action["None"] = "none";
    Action["GroupRant"] = "grouprant";
    Action["Rant"] = "rant";
})(Action = exports.Action || (exports.Action = {}));
var CollabState;
(function (CollabState) {
    CollabState[CollabState["Unknown"] = 0] = "Unknown";
    CollabState[CollabState["OpenSourceProject"] = 1] = "OpenSourceProject";
    CollabState[CollabState["ExistingOpenSourceProject"] = 2] = "ExistingOpenSourceProject";
    CollabState[CollabState["ProjectIdea"] = 3] = "ProjectIdea";
    CollabState[CollabState["ExistingProject"] = 4] = "ExistingProject";
})(CollabState = exports.CollabState || (exports.CollabState = {}));
var VoteState;
(function (VoteState) {
    VoteState[VoteState["Unvoted"] = 0] = "Unvoted";
    VoteState[VoteState["Upvoted"] = 1] = "Upvoted";
    VoteState[VoteState["Downvoted"] = -1] = "Downvoted";
    VoteState[VoteState["NotAllowed"] = -2] = "NotAllowed";
})(VoteState = exports.VoteState || (exports.VoteState = {}));
